common/tree/script.go

```diff
@@ -44,7 +44,7 @@ var forbiddenOpcodes = []byte{
 
 type Closure interface {
 	Script() ([]byte, error)
-	Decode(script []byte) (bool, error)
```

Comment:
I would keep the interface unchanged and instantiate ScriptTokenizer in the function

Reply:
That would make it difficult to use the `cursor` of `Tokenizer` type, which is quite splendid if a Decode type calls another Decode Type.